src/passes/flattening/ControlFlowFlattening.cpp

```diff
@@ -104,6 +104,20 @@ void EmitDefaultCaseAssembly(IRBTy& IRB, Triple TT) {
       /* hasSideEffects */ true,
       /* isStackAligned */ true
     ));
+  } else if (TT.isARM()) {
+    IRB.CreateCall(FType, InlineAsm::get(
+      FType,
+      R"delim(
+        ldr r1, [r1, #-8];
```

Comment:
`ldr x1, #-8;` is actually a pc-relative instruction. Thus, it should be: `ldr r1, [pc, #-4];`

Reply:
Definitely, thanks for catching it!